python/VoxelMapper.py

```diff
@@ -1,28 +1,121 @@
+#!/usr/bin/env python
 import sys
 import os
 import nibabel as nib
 import  numpy as np
+import numbers
+import argparse
 
-for line in sys.stdin:
-	line = line.strip()
-	slice_file = os.path.join(line)
-	
-        #open nifti image
-        slice = nib.load(slice_file)
-	data = slice.get_data().flat
- 
-        histogram = {}
-	    
-        #iterate throught the voxels and add them to dictionary with an initial count of 1
-        #if same voxel value is encountered again, increment the count of the voxel value by one
-	for key in data:
+
+        
+def mapper(bin_size, data_max, data_min):
+    for line in sys.stdin:
+            line = line.strip()
+            line = line.split()
+            slice_file = os.path.join(line[0])
+            
+
+            #open nifti image
+            slice = nib.load(slice_file)
+            data = slice.get_data().flat
+     
+            histogram = {}
+            
+
+            #iterate throught the voxels and add them to dictionary with an initial count of 1
+            #if same voxel value is encountered again, increment the count of the voxel value by one
+            for value in data:
+                try:
+                    voxel_value = value.item()
+                except:
+                    print('Error: Voxel value could not be cast to a python scalar')
+                    sys.exit(0)
+                
+                #dictionary key will be in the form of (min, max) tuple. All values will be in min/max range, excluding the max, which will be the min value of the proceeding bin
+                if isinstance(value, numbers.Number):
+                    
+                    if data_min is not None and data_max is not None and (voxel_value < data_min or voxel_value >= data_max):
```

Comment:
If I understand well, data_min and data_max have to be defined before the method is called. So the method could assume they're not `None` and save some logic.